api/apis/space_manifest_handler.go

```diff
@@ -34,19 +36,22 @@ func NewSpaceManifestHandler(
 	logger logr.Logger,
 	serverURL url.URL,
 	applyManifestAction ApplyManifestAction,
+	spaceRepo CFSpaceRepository,
 	buildClient ClientBuilder,
 	k8sConfig *rest.Config) *SpaceManifestHandler {
 	return &SpaceManifestHandler{
 		logger:              logger,
 		serverURL:           serverURL,
 		applyManifestAction: applyManifestAction,
+		spaceRepo:           spaceRepo,
 		buildClient:         buildClient,
 		k8sConfig:           k8sConfig,
 	}
 }
 
 func (h *SpaceManifestHandler) RegisterRoutes(router *mux.Router) {
 	router.Path(SpaceManifestApplyEndpoint).Methods("POST").HandlerFunc(h.applyManifestHandler)
+	router.Path(SpaceManifestDiffEndpoint).Methods("POST").HandlerFunc(h.validateSpaceVisible(h.diffManifestHandler))
```

Comment:
Eventually we'd like to apply this same check to the `applyManifestHandler`, which is why I made the wrapping handler, but it was expensive to update its integration tests due to the coupling between Spaces and HNS resources and HNC. It's out of scope for this story (which is a blocker for `cf push` to proceed) so I decided to skip it for now, but I feel like this is pointing at another issue with the tight coupling between orgs/spaces and HNC.